src/System.Text.Json/porting_guide/README.md

```diff
@@ -0,0 +1,173 @@
+  - Unlike searching and accessing JSON elements on `JObject`/`JArray`, which tend to be relatively fast lookups in some dictionary, searching and accessing JSON elements on `JsonElement` requires sequential lookup (since we trade-off initial parse time for look-up time). Therefore, it is recommended that you try to avoid searching the whole `JsonDocument` for every property, but rather search on nested JSON objects that you already have as `JsonElement`. For the same reason, we encourage you to use the built-in enumerators (`EnumerateArray` and `EnumerateObject`) rather than doing your own indexing/loops.
+
+```C#
+static double ParseJson()
+{
+    const string json = " [ { \"name\": \"John\" }, [ \"425-000-1212\", 15 ], { \"grades\": [ 90, 80, 100, 75 ] } ]";
+
+    double average = -1;
+
+    using (JsonDocument doc = JsonDocument.Parse(json))
+    {
+        JsonElement root = doc.RootElement;
+        JsonElement info = root[1];
+
+        string phoneNumber = info[0].GetString();
+        int age = info[1].GetInt32();
+
+        JsonElement grades = root[2].GetProperty("grades");
+
+        double sum = 0;
+        foreach (JsonElement grade in grades.EnumerateArray())
+        {
+            sum += grade.GetInt32();
+        }
+
+        int numberOfCourses = grades.GetArrayLength();
+        average = sum / numberOfCourses;
+    }
+
+    return average;
+}
+```
+
+### Multi-Targeting Various TFMs
+
+* If possible, you should target .NET Core 3.0 and get the in-box `System.Text.Json` APIs. However, if you need to support netstandard2.0 (for example, if you are a library developer), you can try to use the **unsupported** source package (see guidance [here](https://github.com/dotnet/corefx/tree/master/src/System.Text.Json/source_package)). If, however, you need to target an older platform or standard, or for some other reason would like to continue to use `Newtonsoft.Json` on certain platforms, you can try to multi-target and have two implementations. However, this is not trivial and would require some `#ifdefs` and source duplication especially if you heavily rely on features that only exist in `Newtonsoft.Json`. One pattern to try to share as much code as possible is to create a `ref struct` wrapper around types like `Utf8JsonReader`/`JsonTextReader` and `Utf8JsonWriter`/`JsonTextWriter` to unify the public surface area used while isolating the behavioral differences. This way you can isolate the changes mainly to the construction of the type (along with passing the new type around by ref). In fact, that is the pattern we currently follow in [core-setup](https://github.com/dotnet/core-setup):
```

Comment:
drop "unsupported".  The source package is just as supported as any other corefx nuget package.

Reply:
Sure, will do.

cc @ericstj, @terrajobst